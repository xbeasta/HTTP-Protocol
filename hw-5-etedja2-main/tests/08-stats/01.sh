#!/bin/bash

PORT=$@

REQUEST=$'GET /stats HTTP/1.1\r\n\r\n'
PING=$'GET /ping HTTP/1.1\r\n\r\n'

echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
#curl -s http://127.0.0.1:$PORT/ping &> /dev/null
echo -n "$PING" | nc 127.0.0.1 $PORT &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
#curl -s http://127.0.0.1:$PORT/ping &> /dev/null
echo -n "$PING" | nc 127.0.0.1 $PORT &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/read &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/echo &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/05-file/threads.txt &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-files/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-error/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
curl -s http://127.0.0.1:$PORT/tests/06-error/index.html &> /dev/null
echo -n "$REQUEST" | nc 127.0.0.1 $PORT
echo ""
