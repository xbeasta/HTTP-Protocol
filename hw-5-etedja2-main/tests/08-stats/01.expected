HTTP/1.1 200 OK
Content-Length: 66

Requests: 0
Header bytes: 0
Body bytes: 0
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 68

Requests: 1
Header bytes: 39
Body bytes: 66
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 69

Requests: 2
Header bytes: 78
Body bytes: 134
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 3
Header bytes: 117
Body bytes: 203
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 4
Header bytes: 156
Body bytes: 273
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 6
Header bytes: 233
Body bytes: 347
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 70

Requests: 8
Header bytes: 311
Body bytes: 476
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 10
Header bytes: 388
Body bytes: 550
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 12
Header bytes: 466
Body bytes: 680
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 14
Header bytes: 544
Body bytes: 810
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 71

Requests: 16
Header bytes: 621
Body bytes: 888
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 18
Header bytes: 699
Body bytes: 1018
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 72

Requests: 20
Header bytes: 776
Body bytes: 1091
Errors: 0
Error bytes: 0
HTTP/1.1 200 OK
Content-Length: 73

Requests: 21
Header bytes: 815
Body bytes: 1163
Errors: 1
Error bytes: 26
HTTP/1.1 200 OK
Content-Length: 73

Requests: 22
Header bytes: 854
Body bytes: 1236
Errors: 2
Error bytes: 52
HTTP/1.1 200 OK
Content-Length: 73

Requests: 23
Header bytes: 893
Body bytes: 1309
Errors: 3
Error bytes: 78
